pkg/pillar/cmd/zedrouter/zedrouter.go

```diff
@@ -2616,3 +2663,167 @@ func handleDNSDelete(ctxArg interface{}, key string,
 	maybeHandleDNS(ctx)
 	log.Infof("handleDNSDelete done for %s\n", key)
 }
+
+func validateAppNetworkConfig(ctx *zedrouterContext, appNetConfig types.AppNetworkConfig,
+	appNetStatus *types.AppNetworkStatus) bool {
+	log.Infof("AppNetwork(%s), check for duplicate port map acls", appNetConfig.DisplayName)
+	// For App Networks, check for common port map rules
+	ulCfgList0 := appNetConfig.UnderlayNetworkList
+	if len(ulCfgList0) == 0 {
+		return false
+	}
+	sub := ctx.subAppNetworkConfig
+	items := sub.GetAll()
+	for _, cfg := range items {
+		appNetConfig1 := cast.CastAppNetworkConfig(cfg)
+		ulCfgList1 := appNetConfig1.UnderlayNetworkList
+		if appNetConfig.DisplayName == appNetConfig1.DisplayName ||
+			len(ulCfgList1) == 0 {
+			continue
+		}
+		ret, err := checkUnderlayNetworkForDuplicatePortMap(ctx, ulCfgList0,
+			ulCfgList1)
+		if err != nil {
```

Comment:
For all ACL rules we get errors at some point. Why should we check for errors a second time for portmap rules? We just need to add the portmap overlap check. and if there is no NetworkInstanceConfig/Status there is no overlap.

Reply:
sure, done.